pkgs/development/python-modules/pyfatfs/default.nix

```
@@ -0,0 +1,51 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  setuptools-scm,
+  fs,
+  pytestCheckHook,
+  pytest-mock,
+}:
+
+buildPythonPackage rec {
+  pname = "pyfatfs";
+  version = "1.1.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "nathanhi";
+    repo = "pyfatfs";
+    tag = "v${version}";
+    hash = "sha256-26b4EV3WERUqJ10VkYov3PDFhSBcfxCF79P8Eg5xpoM=";
+  };
+
+  build-system = [
+    setuptools
+    setuptools-scm
+  ];
+
+  dependencies = [ fs ];
+
+  postPatch = ''
+    substituteInPlace ./pyproject.toml \
+      --replace-fail 'setuptools ~= 67.8' setuptools \
+      --replace-fail '"setuptools_scm[toml] ~= 7.1"' ""
+  '';
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-mock
+  ];
+
+  env.SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  meta = {
+    description = "Python based FAT12/FAT16/FAT32 implementation with VFAT support";
+    homepage = "https://github.com/nathanhi/pyfatfs";
+    license = lib.licenses.mit;
+    platform = lib.platforms.all;
```

See [CI failure](https://github.com/NixOS/nixpkgs/actions/runs/12253963059/job/34183933678)
```suggestion
    platforms = lib.platforms.all;
```

---

yep,sorry for the typo